### PR TITLE
CRUD

### PR DESCRIPTION
**Pull request by a-stasevich**

### CODE REVIEW

File: core/client/api/content.js [hunk 1]
```diff
@@ -7,6 +7,16 @@ const load = async (contentType, slug) => {
   return response.data;
 };
 
+const create = async (contentType, slug, data) => {
+  const response = await axios.post(`${HOST}/_api/content/${contentType}/${slug}/new`, data);
```
Comment (sergesemashko) [inline core/client/api/content.js]: you can omit `${slug}/new` and just do post request to server passing the slug in the body. Post to `_api/content/${contentType}` would be sufficient in REST to tell server to create something. you can read more here about REST API overview - https://medium.com/the-coding-matrix/rest-101-ea5e54e3290c

File: server/content.js [hunk 2]
```diff
@@ -2,14 +2,56 @@ const fse = require('fs-extra');
...
+      if (err === null) {
+        throw fileExistError;
+      } else if (err.code === 'ENOENT') {
+        fs.rename(filePath, newFilePath, err => {
+          if (err) throw err;
+          fse.outputFile(newFilePath, data, cb);
+        });
+      } else {
+        console.error(err.code);
+        throw err;
```
Comment (sergesemashko) [inline server/content.js:24]: thrown errors should be handled on the API level as well to pass related HTTP error code back to browser. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors

File: server/content.js [hunk 3]
```diff
@@ -2,14 +2,56 @@ const fse = require('fs-extra');
...
+    fs.stat(newFilePath, function(err, stat) {
+      if (err === null) {
+        throw fileExistError;
+      } else if (err.code === 'ENOENT') {
+        fs.rename(filePath, newFilePath, err => {
+          if (err) throw err;
+          fse.outputFile(newFilePath, data, cb);
+        });
+      } else {
+        console.error(err.code);
```
Comment (sergesemashko) [inline server/content.js]: do you need console here?

File: server/index.js [hunk 4]
```diff
@@ -37,14 +37,32 @@ app.prepare().then(() => {
...
       JSON.stringify(req.body, null, 2),
       data => {
-        res.send('');
+        res.send(JSON.stringify(data));
+      },
+    );
+  });
+
+  server.post('/_api/content/:type/:slug/new', (req, res) => {
+    console.log(req.body);
```
Comment (sergesemashko) [inline server/index.js]: let's remove console.logs from this file

File: server/content.js (hunk 2 above)
Comment (a-stasevich) [reply server/content.js:24]: So which one should we use here ?

File: server/content.js (hunk 3 above)
Comment (a-stasevich) [reply server/content.js]: I didn't

**Comment by sergesemashko** [comment]
I'm having following error when trying to save a new content:
```

/Users/sis/Sites/sergesemashko/flatlify/server/content.js:37
      throw fileExistError;
      ^

Error: [object Object]
    at Object.<anonymous> (/Users/sis/Sites/sergesemashko/flatlify/server/content.js:5:24)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
    at loader (/Users/sis/Sites/sergesemashko/flatlify/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/sis/Sites/sergesemashko/flatlify/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:685:32)
    at Function.Module._load (internal/modules/cjs/loader.js:620:12)
    at Module.require (internal/modules/cjs/loader.js:723:19)
    at require (internal/modules/cjs/helpers.js:14:16)
    at Object.<anonymous> (/Users/sis/Sites/sergesemashko/flatlify/server/index.js:9:17)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
[nodemon] app crashed - waiting for file changes before starting...
```
- looks like there is an error on `Delete content` click too, as nothing happens

**Comment by sergesemashko** [comment]
@a-stasevich, let's don't fix anything yet. I realized from the exercise we were going through, that we should be building an admin interface. I figured there are options out there: [crudl](https://crudl.io/), https://www.forestadmin.com (really good one, but in Ruby which makes it complicated).

cc/ @JeminiBo

**Pull request merged**